modules/frontend/combiner/metrics_query_range.go

```diff
@@ -58,7 +58,9 @@ func NewQueryRange(req *tempopb.QueryRangeRequest, maxSeriesLimit int) (Combiner
 			if combiner.MaxSeriesReached() {
 				// Truncating the final response because even if we bail as soon as len(resp.Series) >= maxSeries
 				// it's possible that the last response pushed us over the max series limit.
-				resp.Series = resp.Series[:maxSeries]
+				if len(resp.Series) > maxSeries {
```

Comment:
Does this also need to check `&& maxSeries > 0` to handle the case of unlimited?  Not sure if there is a case where we could get `combiner.MaxSeriesReached()` and maxSeries==0.